catalog/spatial/registry/registry-admin-modules/registry-admin-remote-ui/src/main/webapp/js/view/Registry.view.js

```diff
@@ -195,14 +196,33 @@ function (ich,Marionette,_,$,Q,ModalRegistry,EmptyView,wreqr,Utils,deleteRegistr
             });
         },
         addRegistry: function(){
-            if(this.model) {
-                wreqr.vent.trigger("showModal",
-                    new ModalRegistry.View({
-                        model: this.model.getRegistryModel(),
-                        registry: this.model,
-                        mode: 'add'
-                    })
-                );
+            var self = this.model;
+            var configNames = self.getRegistryModel().get('configNames');
+            var regConfigs = self.getRegistryModel().get('registryConfiguration');
+            regConfigs.forEach(function(config){
+                var service = config.get('service').get('name');
+                if(configNames.indexOf(service) === -1){
+                    configNames.push(service);
+                }
+            });
```

Comment:
Refactored, I placed it here originally since I thought it was the closest to the modal that would ensure the array is populated before creation. Not sure why the initialize function of the modal wasn't fast enough, but ... it is better now.